### PR TITLE
Game authorization score migration

### PR DESCRIPTION
**Pull request by olafpena**

### CODE REVIEW

File: game-authorization-score/src/main/java/com/iconbet/score/authorization/Authorization.java [hunk 1]
```diff
@@ -0,0 +1,965 @@
...
+		//Context.getOrigin() - > txn.origin  - always wallet
+		//Context.getCaller() - > sender
+		//Context.getOrigin() - > owner
+		if (!Context.getOrigin().equals(Context.getOwner()))
+			Context.revert("Only the owner can call the untether method.");
+	}
+	
+	
+	@External
+	public void set_new_div_changing_time(BigInteger timestamp) {
```
Comment (cacomixtle) [inline game-authorization-score/src/main/java/com/iconbet/score/authorization/Authorization.java:166]: Not sure if we need to preserve the parameter name, we need to test it.
lets say if we convert it to http request... 

http://localhost/set_new_div_changing_time?timestamp=10000000000000000 
is different than
http://localhost/set_new_div_changing_time?_timestamp=10000000000000000

File: game-authorization-score/src/main/java/com/iconbet/score/authorization/Authorization.java [hunk 2]
```diff
@@ -0,0 +1,965 @@
...
+		Address sender = Context.getCaller();
+		
+		if (!sender.equals(this.roulette_score.get()) ) {
+			Context.revert("Only roulette score can invoke this method.");			
+		}
+		BigInteger now = BigInteger.valueOf(Context.getBlockTimestamp());
+		BigInteger day = now.divide(U_SECONDS_DAY);
+		
+		//Question ? day.intValue() number of the day
+		BigInteger wagerValue = this.wagers.get(game)[day.intValue()];
```
Comment (cacomixtle) [inline game-authorization-score/src/main/java/com/iconbet/score/authorization/Authorization.java:516]: you have a lot of null pointer exception issues
you init the dictdb but has not init the BigInteger[] 
if this.wagers.get(game) returns null then accesing [day.iniValue()] will throws nullpointerex

same case for variables: 
payouts
games_excess_history

you need to verify
1) if the method is writable @External then 
```
if (this.wagers.get(game) == null ) 
      this.wagers.set(game, new BigInteger[]{ZERO,ZERO,ZERO,..., <number of enums in the structure>}) 
// use this.wagers.get(game)[X]
```

2) if the method is read only @External(readOnly=true)
```
BigInteger[] list = new BigInteger[]{ZERO,ZERO,ZERO,..., <number of enums in the structure>};
if (this.wagers.get(game) != null ) 
     list= this.wagers.get(game)

//use ... list[X]...

```       
if the array is too big, replace BigInteger[] by LinkedList<BigInteger> so you just use the necessary space

File: game-authorization-score/src/main/java/com/iconbet/score/authorization/Authorization.java [hunk 3]
```diff
@@ -0,0 +1,965 @@
...
+import score.DictDB;
+import score.VarDB;
+import score.annotation.EventLog;
+import score.annotation.External;
+import score.annotation.Payable;
+
+import com.iconloop.score.token.irc2.IRC2;
+
+
+public class Authorization implements IRC2{
```
Comment (cacomixtle) [inline game-authorization-score/src/main/java/com/iconbet/score/authorization/Authorization.java:25]: this is not a Token, this is just a contract, do not implement IRC2
The class in py does not inherits from a Token

class Authorization(IconScoreBase):

TokenStandard = IRC2.class

File: game-authorization-score/src/main/java/com/iconbet/score/authorization/Authorization.java [hunk 4]
```diff
@@ -0,0 +1,965 @@
...
+	private final VarDB<BigInteger> new_div_changing_time = Context.newVarDB(NEW_DIV_CHANGING_TIME, BigInteger.class);
+	private final DictDB<Address, BigInteger[]> games_excess_history = Context.newDictDB(TODAYS_GAMES_EXCESS, BigInteger[].class);
+
+	private final VarDB<Boolean> apply_watch_dog_method = Context.newVarDB(APPLY_WATCH_DOG_METHOD, Boolean.class);
+	private final DictDB<Address,BigInteger> maximum_payouts = Context.newDictDB(MAXIMUM_PAYOUTS, BigInteger.class);
+	private final VarDB<BigInteger> maximum_loss = Context.newVarDB(MAXIMUM_LOSS, BigInteger.class);
+
+	  
+
+	@Override
```
Comment (cacomixtle) [inline game-authorization-score/src/main/java/com/iconbet/score/authorization/Authorization.java:113]: remove the methods inherited from TokenStandard IRC2

**Review by cacomixtle — CHANGES_REQUESTED** [review]

File: game-authorization-score/src/main/java/com/iconbet/score/authorization/Authorization.java (hunk 2 above)
Comment (olafpena) [reply game-authorization-score/src/main/java/com/iconbet/score/authorization/Authorization.java]: updated accordingly.

File: game-authorization-score/src/main/java/com/iconbet/score/authorization/Authorization.java (hunk 1 above)
Comment (olafpena) [reply game-authorization-score/src/main/java/com/iconbet/score/authorization/Authorization.java]: Updated variable names with original names.

File: game-authorization-score/src/main/java/com/iconbet/score/authorization/Authorization.java (hunk 3 above)
Comment (olafpena) [reply game-authorization-score/src/main/java/com/iconbet/score/authorization/Authorization.java]: IRC2 interface removed and methods as well

File: game-authorization-score/src/main/java/com/iconbet/score/authorization/Authorization.java (hunk 4 above)
Comment (olafpena) [reply game-authorization-score/src/main/java/com/iconbet/score/authorization/Authorization.java]: removed.

**Review by cacomixtle — APPROVED** [review]

**Review by ismaelmk — APPROVED** [review]

**Pull request merged**